lib/linked_list.rb

```diff
@@ -18,132 +18,262 @@ def initialize
 
     # method to add a new node with the specific data value in the linked list
     # insert the new node at the beginning of the linked list
-    # Time Complexity:  
-    # Space Complexity
+    # Time Complexity: O(1) 
+    # Space Complexity: O(1)
     def add_first(value)
-      raise NotImplementedError
+      new_node = Node.new(value)
+
+      new_node.next = @head if @head

```

Comment:
Do you need the `if` statement here?  Even if `@head` is `nil` you can assign next to it.